### PR TITLE
Install harfbuzz and point to public headers directory

### PR DESCRIPTION
**Pull request by ebraminio**
Also fixes a minor issue with release fetching that compiles the port every time on bz2 files.

### CODE REVIEW

File: tools/system_libs.py [hunk 1]
```diff
@@ -647,10 +647,7 @@ def check_tag():
       else:
         names = zipfile.ZipFile(fullname + '.zip', 'r').namelist()
 
-      if not (names[0].startswith(subdir + '/') or names[0].startswith(subdir + '\\')):
-        # current zip file is old, force a retrieve
-        return False
-      return True
+      return bool(re.match(subdir + r'(\\|/|$)', names[0]))
```
Comment (kripken) [inline tools/system_libs.py]: I don't understand what this change does? Perhaps add a comment in the source too?

Comment (ebraminio) [reply tools/system_libs.py]: Separate this part of the patch to #6273, I am not sure why bots are failing but that part hopefully doesn't have any issue.

**Comment by ebraminio** [comment]
Lets see if bots now like this now and if not, why.

**Comment by kripken** [comment]
Sorry for the delay here. Looks ok on the bots, and the change seems ok to me, although I admit I don't see what was wrong before - maybe we should have a test that checks this? (I assume we don't have one since nothing was broken before this PR?)

**Comment by ebraminio** [comment]
@tn0502 can you see why the bots are failing and maybe picking up PR? I don't have a ready to use emscripten checkout around :(

**Comment by tn0502** [comment]
@ebraminio Have you tried rebasing on top of latest master? I applied your changes to a clean master and the test passed locally:

```
$ EMCC_CORES=4 python tests/runner.py test_emscripten_get_now
WARNING:tests/runner.py:use EM_ALL_ENGINES=1 in the env to run against all JS engines, which is slower but provides more coverage
Test suites:
['test_core']
Running test_core: (1 tests)
test_emscripten_get_now (test_core.binaryen0) ... (checking sanity from test runner)
INFO:root:(Emscripten: Running sanity checks)
ok

----------------------------------------------------------------------
Ran 1 test in 2.434s

OK
```

**Comment by tn0502** [comment]
@ebraminio Btw, I was not able to successfully build my project with this patch. I had to do this instead:

```diff
diff --git a/tools/ports/harfbuzz.py b/tools/ports/harfbuzz.py
index a584794da..13c6ddbd9 100644
--- a/tools/ports/harfbuzz.py
+++ b/tools/ports/harfbuzz.py
@@ -35,7 +35,7 @@ def process_dependencies(settings):
 def process_args(ports, args, settings, shared):
   if settings.USE_HARFBUZZ == 1:
     get(ports, settings, shared)
-    args += ['-Xclang', '-isystem' + os.path.join(ports.get_build_dir(), 'harfbuzz', 'include', 'harfbuzz')]
+    args += ['-Xclang', '-isystem' + os.path.join(ports.get_dir(), 'harfbuzz', 'harfbuzz-' + TAG, 'src')]
   return args

 def show():
```

Which seems to me like there's a missing install step somewhere. I'm not familiar with how this script is executed so I'm unsure where to put it, and what such a step would look like.

**Comment by tn0502** [comment]
@ebraminio I have a PR for the issue with Harfbuzz being downloaded with every compilation unit: #6977. That change in combination with the patch I posted earlier resolves all of my issues.

**Comment by ebraminio** [comment]
Excelllllent, so closing in favor of yours #6977 :)

**Comment by ebraminio** [comment]
Oh, now I see that wasn't equivalent of this but well I guess you can do that :P Also feel free to have a look at https://github.com/kripken/emscripten/pull/6270 I tried to update freetype port there which I guess will useful for your work :)

**Comment by tn0502** [comment]
I've published my patch here: #6978. We'll see if I have time for #6270 …

**Pull request closed without merging**